internal/integration/infra_sdk_emitter.go

```diff
@@ -122,11 +120,66 @@ func (e *InfraSdkEmitter) emitSummary(i *integration.Integration, metric Metric,
 		ps.AddQuantile(*q.Quantile, *q.Value)
 	}
 
-	i.HostEntity.AddMetric(ps)
+	return e.addMetricToEntity(i, metric, ps)
+}
+
+func (e *InfraSdkEmitter) addMetricToEntity(i *integration.Integration, metric Metric, m metrics.Metric) error {
+	entityProps, err := e.definitions.getEntity(metric)
+	// if we can't find an entity for the metric, add it to the "host" entity
+	if err != nil {
+		logrus.WithError(err).Debugf("failed to map metric to entity. using 'host' entity")
+		i.HostEntity.AddMetric(m)
+		return nil
+	}
+
+	entityName := buildEntityName(entityProps, m)
+	// try to find the entity and add the metric to it
+	// if there's no entity with the same name yet, create it and add it to the integration
+	entity, ok := i.FindEntity(entityName)
+	if !ok {
+		entity, err = i.NewEntity(entityName, entityProps.Type, entityName)
+		if err != nil {
+			logrus.WithError(err).Errorf("failed to create entity %v", entityName)
+			return err
+		}
+		i.AddEntity(entity)
+	}
 
+	entity.AddMetric(m)
 	return nil
 }
 
+// build the entity name based on various properties
+// the format should be as follows:
+//  serviceName:exporterHost:exporterPort:entityName:dimension1:dimension2..
```

Comment:
is there a limit for the length of the entity name we can hit?

Reply:
good point. I think there is. we should confirm this with platform

Reply:
Platform has told me that it's 500 chars. So we should perhap introduce a limit here around that value.
I see an issue where if we remove some dimensions from the name it can lead to the entity name not being unique.